provisioning/device/src/Models/DeviceRegistrationResult.cs

```diff
@@ -115,7 +115,7 @@ public bool TryGetPayload<T>(out T value)
 
             try
             {
-                value = Payload.Value<T>();
```

Comment:
This is incorrect and would cause `InvalidCastException` while deserializing to some particular type of objects. Correcting as referring to [here](https://github.com/Azure/azure-iot-sdk-csharp/blob/previews/v2/iothub/service/src/DirectMethod/Models/DirectMethodClientResponse.cs#L59).